docs/api.md

```diff
@@ -535,7 +535,7 @@ When `spec.nodeReuse` field of metal3MachineTemplate is set to `True`, CAPM3
 Machine controller:
 
 - Sets `infrastructure.cluster.x-k8s.io/node-reuse` label to the corresponding
-  CAPI object name (`KubeadmControlPlane` or `MachineDeployment`) on the
+  CAPI object name (a `controlplane.cluster.x-k8s.io` object such as `KubeadmControlPlane` or a `MachineDeployment`) on the
```

Comment:
ah yes, indeed

the text could capture the `cp-` or `md-` prefix though

```suggestion
  CAPI object name (a `controlplane.cluster.x-k8s.io` object such as `KubeadmControlPlane` or a `MachineDeployment`), prefixed by `cp-` or`md-` (to distinguish control plane and machine deployments), on the
```

Reply:
That is a good point. It seems we have no protection from name collisions currently. I would suggest a separate PR for this though since it would be more of a change in behavior. We will need to consider upgrade scenarios also if we change it (probably just document that CAPM3 should not be upgraded at the same time as the MD or CP is rolling out).

Reply:
I think the behaviour in the event of a ControlPlane/MachineDeployment name collision is well defined, the [controlplane is checked first so will take precedence](https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/baremetal/metal3machine_manager.go#L972), we could perhaps just document that to avoid any change in behaviour?